app/src/main/java/com/wireguard/android/services/NotificationService.kt

```diff
@@ -0,0 +1,84 @@
+package com.wireguard.android.services
+
+import android.app.IntentService
+import android.app.Notification
+import android.app.PendingIntent
+import android.content.Intent
+import android.os.Handler
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.wireguard.android.R
+import com.wireguard.android.activity.MainActivity
+import com.wireguard.android.model.TunnelManager
+import timber.log.Timber
+import java.text.SimpleDateFormat
+import java.util.*
+
+class NotificationService: IntentService("Notification Service") {
+
+    private var name: String? = null
+    private var id: Int? = null
+    private val handler = Handler()
+    private val timer = Timer()
+    private lateinit var notificationManager: NotificationManagerCompat
+    private lateinit var builder: NotificationCompat.Builder
+
+
+    override fun onHandleIntent(intent: Intent?) {
+        id = intent?.getIntExtra("id", 0)
+        name = intent?.getStringExtra("name")
+        notificationManager = NotificationManagerCompat.from(this)
+        builder = createNotificationBuilder(name!!)
+        startForeground(id!!, builder.build())
+        updateNotifications()
+    }
+
+    private fun updateNotifications() {
+        val builder = createNotificationBuilder(name!!)
+        val dateFormat = SimpleDateFormat("HH:mm:ss", Locale.getDefault())
```

Comment:
What I think #155 really asked for is a timer that starts when the tunnel goes up, so essentially a counter that goes up in seconds and then auto-formats in hours, minutes and seconds.